src/javascript/tests/functions.rs

```diff
@@ -0,0 +1,56 @@
+use crate::assert_js;
+
+#[test]
+fn exported_functions() {
+    assert_js!(
+        r#"
+pub fn add(x, y) {
+    x + y
+}"#,
+        r#""use strict";
+
+export function add(x, y) {
+  return x + y;
+}
+"#
+    );
+}
+
+#[test]
+fn calling_functions() {
+    assert_js!(
```

Comment:
Can we have some tests that cover the formatting? Would like to see tests that show show the code is split over multiple lines due to

- argument names too long
- function names too long
- nested function names too long
- function body too long
- a sequence of expressions in the function body

Reply:
Not sure what you mean by nested function names two long

Reply:
Sorry, not very clear. Kinda like this:

```rust
pub fn main() {
  aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa(
    bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb(
      1111111111111,
      2222222222222222,
    )
  )
}